esp/esdllib/esdl_script.cpp

```diff
@@ -0,0 +1,919 @@
+            return true;
+        bool match = false;
+        try
+        {
+            match = xpathContext->evaluateAsBoolean(m_test);
+        }
+        catch (IException* e)
+        {
+            int code = e->errorCode();
+            StringBuffer msg;
+            e->errorMessage(msg);
+            e->Release();
+            esdlOperationError(code, tagname, msg, !ignoreCodingErrors);
+        }
+        catch (...)
+        {
+            VStringBuffer msg("unknown exception evaluating test '%s'", m_test.get() ? m_test->getXpath() : "undefined!");
+            esdlOperationError(ESDL_SCRIPT_Error, tagname, msg, !ignoreCodingErrors);
+        }
+        return match;
+    }
+};
+
+class CEsdlTransformOperationChoose : public CEsdlTransformOperationWithChildren
+{
+public:
+    CEsdlTransformOperationChoose(IPropertyTree * tree, const StringBuffer &prefix) : CEsdlTransformOperationWithChildren(tree, prefix, false)
+    {
+        if (tree)
+        {
+            loadWhens(tree, prefix);
+            loadOtherwise(tree, prefix);
+        }
+    }
+
+    virtual ~CEsdlTransformOperationChoose(){}
+
+    bool process(IEspContext * context, IPropertyTree *content, IXpathContext * xpathContext) override
+    {
+        return processChildren(context, content, xpathContext);
+    }
+
+    virtual bool processChildren(IEspContext * context, IPropertyTree *content, IXpathContext * xpathContext) override
+    {
+        ForEachItemIn(i, m_children)
+        {
+            if (m_children.item(i).process(context, content, xpathContext))
+                return true;
+        }
+        return false;
+    }
+
+    virtual void toDBGLog () override
+    {
+    #if defined(_DEBUG)
+        DBGLOG (">>>>>>>>>>> %s >>>>>>>>>>", tagname.str());
+        CEsdlTransformOperationWithChildren::toDBGLog();
+        DBGLOG (">>>>>>>>>>> %s >>>>>>>>>>", tagname.str());
+    #endif
+    }
+
+protected:
+    void loadWhens(IPropertyTree * tree, const StringBuffer &prefix)
+    {
+        StringBuffer xpath;
+        Owned<IPropertyTreeIterator> children = tree->getElements(makeOperationTagName(xpath, prefix, "when"));
+        ForEach(*children)
+            m_children.append(*new CEsdlTransformOperationConditional(&children->query(), prefix));
+    }
+
+    void loadOtherwise(IPropertyTree * tree, const StringBuffer &prefix)
+    {
+        StringBuffer xpath;
+        IPropertyTree * otherwise = tree->queryPropTree(makeOperationTagName(xpath, prefix, "otherwise"));
+        if (!otherwise)
+            return;
+        m_children.append(*new CEsdlTransformOperationConditional(otherwise, prefix));
+    }
+    virtual void loadChildren(IPropertyTree * tree, const StringBuffer &prefix, bool withVariables) override
+    {
+        loadWhens(tree, prefix);
+        loadOtherwise(tree, prefix);
+    }
+};
+
+IEsdlTransformOperation *createEsdlTransformOperation(IPropertyTree *element, const StringBuffer &prefix)
```

Comment:
Shouldn't we be consistent in our enforcement of case-sensitivity?

Reply:
Good catch, I think I started prototyping with insensitve compares and forgot to change them to strict case sensitive compares.